src/index.ts

```diff
@@ -762,7 +762,7 @@ class Chaperone {
 			curr_state	= 2;
 		    if ( this.opened === true )	// setting up wormhole
 			curr_state	= 3;
-		    if ( this.anonymous === true || this.wormhole_ready === true ) // ready
+		    if ( this.opened === true  || (this.anonymous === true || this.wormhole_ready === true) ) // ready
```

Comment:
This change will make state 4 always overwrite state 3.  Did you mean?
```suggestion
		    if ( this.opened === true  && (this.anonymous === true || this.wormhole_ready === true) ) // ready
```

Reply:
:sweat_smile: Yes.